spec/unit/plugins/solaris2/filesystem.rb

```diff
@@ -33,51 +33,289 @@
+        data0/Movies    compression     off     inherited from data0
+        data0/Movies    atime   off     local
+        data0/Movies    devices on      default
+        data0/Movies    exec    on      default
+        data0/Movies    setuid  on      default
+        data0/Movies    readonly        off     default
+        data0/Movies    zoned   off     default
+        data0/Movies    snapdir hidden  default
+        data0/Movies    aclinherit      restricted      default
+        data0/Movies    canmount        on      default
+        data0/Movies    xattr   on      default
+        data0/Movies    copies  1       default
+        data0/Movies    version 5       -
+        data0/Movies    utf8only        off     -
+        data0/Movies    normalization   none    -
+        data0/Movies    casesensitivity sensitive       -
+        data0/Movies    vscan   off     default
+        data0/Movies    nbmand  off     default
+        data0/Movies    sharesmb        off     local
+        data0/Movies    refquota        0       default
+        data0/Movies    refreservation  0       default
+        data0/Movies    primarycache    all     default
+        data0/Movies    secondarycache  all     default
+        data0/Movies    usedbysnapshots 56115388960     -
+        data0/Movies    usedbydataset   586525311344    -
+        data0/Movies    usedbychildren  0       -
+        data0/Movies    usedbyrefreservation    0       -
+        data0/Movies    logbias latency default
+        data0/Movies    dedup   off     default
+        data0/Movies    mlslabel        none    default
+        data0/Movies    sync    standard        default
+        data0/Movies    refcompressratio        1.00x   -
+        data0/Movies    written 197038505024    -
+        data0/Movies    logicalused     642795343360    -
+        data0/Movies    logicalreferenced       586666194432    -
+        data0/Movies    filesystem_limit        18446744073709551615    default
+        data0/Movies    snapshot_limit  18446744073709551615    default
+        data0/Movies    filesystem_count        18446744073709551615    default
+        data0/Movies    snapshot_count  18446744073709551615    default
+        data0/Movies    snapdev hidden  default
+        data0/Movies    acltype off     default
+        data0/Movies    context none    default
+        data0/Movies    fscontext       none    default
+        data0/Movies    defcontext      none    default
+        data0/Movies    rootcontext     none    default
+        data0/Movies    relatime        off     default
+        data0/Movies    redundant_metadata      all     default
+        data0/Movies overlay off default:
+ZFS_GET
     end
 
     after(:each) do
       Ohai.config[:plugin] = @original_plugin_config
     end
 
-    context "when 'zfs get' properties are not configured" do
-      it "collects all filesystem properties" do
-        expect(plugin).to receive(:shell_out)
+    context "handles zfs properties" do
```

Comment:
If someone wants to give me `df -Pka` and `mount` output from a Solaris box, I'll write some unittests for that part of the code since whoever wrote this originally never did.

@scotthain - you have access to the internal slowlaris boxes, right?

Reply:
Thanks, that works. I also need `df -na` to test that bit too. other tests added.